x-pack/plugins/triggers_actions_ui/public/application/lib/rule_api/bulk_delete.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { HttpSetup } from '@kbn/core/public';
+import { KueryNode } from '@kbn/es-query';
+import { INTERNAL_BASE_ALERTING_API_PATH } from '../../constants';
+import { BulkDeleteResponse } from '../../../types';
+
+export const bulkDeleteRules = async ({
+  filter,
+  ids,
+  http,
+}: {
+  filter?: KueryNode | null;
+  ids?: string[];
+  http: HttpSetup;
+}): Promise<BulkDeleteResponse> => {
+  try {
+    const body = JSON.stringify({
+      ids: ids?.length ? ids : undefined,
+      ...(filter ? { filter: JSON.stringify(filter) } : {}),
```

Comment:
to stringify the filters seems redundant as the initial stringify should stringify everything 🤔

Reply:
I've done like this for consistency with bulk_edit. We have an agreement that we are passing always string from UI as a filter.

Reply:
Sorry there is something I don't understand. Body will be a string because of the first JSON.stringify, stringifying the filter would mean that after having the filters in a string we will run stringify on that string and the only thing that should be different is that " will be  \" and I can't see what the reason behind that might be

Reply:
I think there is two reason, one there is not really away to validate the object of DSL and I also think it is better to see if the stringify will work instead of getting an error on the server side

Reply:
@jcger I agree that it excessive to stringily both filter and body. 
I have only doubts connecting with validation of the filter option on backend. Now it just a string and it's easy to validate. If it'll be KueryNode it would be any or something like this. 
I would not make this change here in this PR. Let me create a ticket where I can change contract for bulk_delete/bulk edit endpoints and make relevant frontend changes. What do you think?

Reply:
At this point I'm not completely following, sounds like we have some complex type checking on the backend but I don't understand what's the matter if we are sending a string no matter what. Also, in case it fails we do nothing but to launch an error, I guess this part would be updated if we start having a KueryNode? Maybe we can discuss it in another occasion, now that you know my concerns and that I don't have the whole picture I'm sure you'll implement whatever makes more sense